src/ec2/model.ts

```diff
@@ -41,14 +41,17 @@ export class Ec2ConnectionManager {
         return new DefaultIamClient(this.regionCode)
     }
 
-    protected async getAttachedPolicies(instanceId: string): Promise<IAM.attachedPoliciesListType> {
+    protected async getAttachedPolicies(instanceId: string): Promise<IAM.AttachedPolicy[]> {
         const IamRole = await this.ec2Client.getAttachedIamRole(instanceId)
         if (!IamRole) {
             return []
         }
-        const iamResponse = await this.iamClient.listAttachedRolePolicies(IamRole!.Arn!)
-
-        return iamResponse.AttachedPolicies ?? []
+        try {
+            const attachedRoles = await this.iamClient.listAttachedRolePolicies(IamRole!.Arn!)
+            return attachedRoles
+        } catch (e) {
+            return []
```

Comment:
does iamClient already log the failure? if not, we should log something here

Reply:
Makes sense, I changed it to log the ARN here.